lib/ThreeScaleClient.php

```diff
@@ -95,7 +93,18 @@ public function getHost() {
    */
   public function authorize($appId, $appKey = null, $serviceId = null, $usage = null) {
     $url = "http://" . $this->getHost() . "/transactions/authorize.xml";
-    $params = array('provider_key' => $this->getProviderKey(), 'app_id' => $appId);
+    $params = array('app_id' => $appId);
+
+    $providerKey = $this->getProviderKey();
+    $serviceToken = $this->getServiceToken();
+
+    if($providerKey) {
+      $params['provider_key'] = $providerKey;
+    }
+
+    if ($serviceToken) {
```

Comment:
I think that is the right way to do it.

I'd consider creating a class for that tuple so it is easier documented. And I'd consider having the api as `new ThreeScaleClientCredentials("service id", "service token")` instead of writing a hash. 

And course it has to be documented that you can pass a service id or the credentials.

👍